sources/web/datalab/auth.ts

```diff
@@ -128,6 +128,17 @@ export function isSignedIn(): boolean {
   return (process.env.DATALAB_ENV != 'local' || fs.existsSync(appCredFile));
 }
 
+function setOauth2Client(parsed_url: any): void {
+  if (!oauth2Client) {
+    var OAuth2:any = google.auth.OAuth2;
+    // TODO(ojarjur): Ideally, we would get the host and port from the parsed_url
+    // rather than hard-coding them. However, the client ID and secret we are using
+    // are limited to localhost only. We should consider making this configurable, or
+    // using the OAuth flow for non-web applications.
+    oauth2Client = new OAuth2(clientId, clientSecret, 'http://localhost:8081/oauthcallback');
```

Comment:
Can we initialize it in the definition https://github.com/GoogleCloudPlatform/datalab/blob/master/sources/web/datalab/auth.ts#L27?